includes/namesilo_packages.php

```diff
@@ -0,0 +1,296 @@
+        $query = $this->Record->select('value')
+            ->from('module_row_meta')
+            ->where('module_row_meta.module_row_id', '=', $module_row_id)
+            ->where('module_row_meta.key', '=', 'packages_map')
+            ->fetch();
+
+        if (!empty($query->value)) {
+            return json_decode($query->value, true);
+        }
+
+        return [];
+    }
+
+    public function savePackagesMap($packages_map, $module_row_id)
+    {
+        $query = $this->Record->select()
+            ->from('module_row_meta')
+            ->where('module_row_meta.module_row_id', '=', $module_row_id)
+            ->where('module_row_meta.key', '=', 'packages_map')
+            ->numResults();
+
+        $fields = [
+            'module_row_id' => $module_row_id,
+            'key' => 'packages_map',
+            'value' => json_encode($packages_map)
+        ];
+
+        if ($query) {
+            return $this->Record->where('module_row_meta.module_row_id', '=', $module_row_id)
+                ->where('module_row_meta.key', '=', 'packages_map')
+                ->update('module_row_meta', $fields);
+        }
+
+        return $this->Record->insert('module_row_meta', $fields);
+    }
+
+    public function getPackagesVars($module_row_id)
+    {
+        $query = $this->Record->select('value')
+            ->from('module_row_meta')
+            ->where('module_row_meta.module_row_id', '=', $module_row_id)
+            ->where('module_row_meta.key', '=', 'packages_vars')
+            ->fetch();
+
+        if (!empty($query->value)) {
+            return json_decode($query->value, true);
+        }
+
+        return [];
+    }
+
+    public function savePackagesVars($vars, $module_row_id)
+    {
+        $query = $this->Record->select()
+            ->from('module_row_meta')
+            ->where('module_row_meta.module_row_id', '=', $module_row_id)
+            ->where('module_row_meta.key', '=', 'packages_vars')
+            ->numResults();
+
+        // Get TLD prices
+        $prices = $this->getPrices();
+
+        foreach ($vars['pricing'] as $tld => $pricing) {
+            foreach ($pricing as $currency => $price_row) {
+                if ($currency !== 'tld') {
+                    $vars['pricing'][$tld][$currency]['previous_registration_price'] = $prices[$tld][$currency]->registration;
+                    $vars['pricing'][$tld][$currency]['previous_renewal_price'] = $prices[$tld][$currency]->renew;
+                }
+            }
+        }
+
+        $fields = [
+            'module_row_id' => $module_row_id,
+            'key' => 'packages_vars',
+            'value' => json_encode($vars)
+        ];
+
+
+        if ($query) {
+            return $this->Record->where('module_row_meta.module_row_id', '=', $module_row_id)
+                ->where('module_row_meta.key', '=', 'packages_vars')
+                ->update('module_row_meta', $fields);
+        }
+
+        return $this->Record->insert('module_row_meta', $fields);
```

Comment:
This function will be rewritten, currently all TLDs are stored in a single JSON object in a single row, instead I will create a row for each TLD. This will make easier to manage the data stored in the database, as storing giant JSON object doesn't seem like an appropriate solution.